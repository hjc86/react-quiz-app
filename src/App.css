* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
  font-family: 'Exo', sans-serif;
  font-weight: 100px;
}

.container {
  display: flex;
  flex-direction: column;
  
}