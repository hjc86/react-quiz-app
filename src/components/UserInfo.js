import React from 'react';


class UserInfo extends React.Component{

    constructor(props) {
        super(props);

        this.state = {
            username: '',
            difficulty: ''
        }
    }

    getUserData = (event) => {
        event.preventDefault();
        this.setState({
            username: event.target[0].value
        })

    }

    handleChange = (event) => {
        this.setState({
            difficulty: event.target.value
        });
    }

    componentDidUpdate() {
        console.log(this.state);
    }
    


    render() {
        
        return( 
            <div>
                <form onSubmit={this.getUserData}>
                    <label>Username:</label>
                    <input type="text" name="username" id="username" onChange={this.handleChange} />

                    <p>Level of Difficulty:</p>
                    <label>Easy</label>
                    <input type="radio" name="difficulty" id="easy" value="easy" checked={this.state.difficulty === "easy"} onChange={this.handleChange} />
                    <label>Medium</label>
                    <input type="radio" name="difficulty" id="medium" value="medium" checked={this.state.difficulty === "medium"} onChange={this.handleChange} />
                    <label>Difficult</label>
                    <input type="radio" name="difficulty" id="difficult" value="difficult" checked={this.state.difficulty === "difficult"} onChange={this.handleChange}/>

                    {/* dropdown for categories */}

                    <input type="submit" />
                </form>
            </div>

        )
    }
}

export default UserInfo;