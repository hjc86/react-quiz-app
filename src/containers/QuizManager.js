import React, { Component } from 'react';
import Quiz from '../components/Quiz';
import Score from '../components/Score';
// import styles from './Q'

class QuizManager extends Component {
    constructor(props){
        super(props)
        
        this.state={
            questionsAnswered: 0,
            playerIndex: 0,
            finished: false,
            quizQA: null,
            score: 0,
            scoreArray: [],
            carryOn: true, 
            winner: null,
            quizFinished: false

        }  
    }
    
    updateAnsweredQuestions=(questionsAnswered)=>{
        this.setState({
            questionsAnswered: questionsAnswered
        })

        if(this.state.questionsAnswered===5 && this.state.playerIndex<this.props.playerInfo.userName.length-1){
            this.setState({
                playerIndex: this.state.playerIndex+1,
                questionsAnswered: 0,
                
            })
        }

        // if(this.state.questionsAnswered===5 && this.state.playerIndex<this.props.playerInfo.userName.length-1){
    
        // }
    }
    
    updateScore= (playerObj)=>{
        if(this.state.questionsAnswered === 4){
            this.setState({
                scoreArray: [...this.state.scoreArray, playerObj.score],
                score: playerObj.score   
             })
        }

        if(this.state.scoreArray.length===this.props.playerInfo.userName.length){
            this.setState({
                carryOn: false
            })
        } 
    } 
    
    render() {
        return (
            <div>
                        {this.state.questionsAnswered === 5?
                        
                            <Score scoreArray={this.state.scoreArray} userName={this.props.playerInfo.userName} /> :
                
                            <Quiz
                            playerInfo={this.props.playerInfo} 
                            currentPlayer={this.props.playerInfo.userName[this.state.playerIndex]} 
                            answeredQuestions={this.updateAnsweredQuestions}
                            score={this.updateScore}/>
                    
                        }

            </div>
        
        )
    }
}


export default QuizManager