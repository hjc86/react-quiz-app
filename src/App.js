import React from 'react';
import UserInfo from './components/UserInfo';
import Quiz from './components/Quiz';
import Score from './components/Score';
import './App.css';


class App extends React.Component {

// states GetUsername, GetAPI, GetQuestions, GetAnswers, GetScore
constructor(props){
  super(props);
  this.state = {
      QuizQ : [],
      Answers : [],
      userName: null,
      difficulty: null,
      category: null
  };
}

// Fetch API
getQuizApi = async (userData) => {

  const categoryNumber = userData.category;
  const difficulty = userData.difficulty;
  const apiCall = await fetch(`https://opentdb.com/api.php?amount=5&category=${categoryNumber}&difficulty=${difficulty}&type=multiple`);
  const data = await apiCall.json();

  console.log(userData);
 
  // to get the whole data 
  this.setState({
    QuizQ : data.results
  
  });
  console.log(data);
}


getUserData =(userDataObj) =>{
  this.setState(
    userDataObj
  ) 
  // console.log("before update",this.state)
}

componentDidUpdate(){
  // console.log("after update",this.state)
}


formSubmitted = (state) => {
  this.setState({
      userName: state.username,
      difficulty: state.difficulty,
      category: state.category
  })
  console.log(state);
  this.getQuizApi(state);
  
}



    render() {
      return (
        <div className="container">
          <UserInfo userData={this.getUserData} onSubmit={this.formSubmitted}/>
          {this.state.QuizQ.map(questions => {
            const answers = [questions.correct_answer, ...questions.incorrect_answers]
            console.log(answers);
            return <Quiz questions={questions.question} answers={questions.correct_answer} incorrect={questions.incorrect_answers}/>
          })
          }
          <Score />
        </div>
      );
}

}

export default App;

// answers={questions.correct_answer}