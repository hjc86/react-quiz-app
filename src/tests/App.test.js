import React, { Component } from 'react';
import ReactDOM from 'react-dom'
import { shallow, mount } from 'enzyme';
import App from '../App';
import renderer from 'react-test-renderer';
import UserInfo from '../components/UserInfo'
import QuizManager from '../containers/QuizManager'


describe(App, () => {
    it('mounts to DOM', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders correctly', () => {
        const tree = renderer
            .create(<App />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    let wrapper;
    beforeEach(() => wrapper = shallow(<App />));
    
    it('Renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('should render a <div />', () => {
        expect(wrapper.find('div').length).toEqual(1);
    });

    it('should render a <UserInfo/>', () => {
        expect(wrapper.find(UserInfo)).toHaveLength(1)
    });

    it('should render a <QuizManager/> if userName state is not null',()=>{
        wrapper.setState({userName: "Tom"})
        expect(wrapper.find(QuizManager)).toHaveLength(1)
    })


})



