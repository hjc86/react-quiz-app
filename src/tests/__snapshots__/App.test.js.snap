// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders correctly 1`] = `ShallowWrapper {}`;

exports[`App renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="formDiv"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        className="player"
      >
        Player : 
      </label>
      <input
        className="bar"
        id="username"
        name="username"
        placeholder="    insert player name(s)"
        required={true}
        type="text"
      />
      <p
        className="diff"
      >
        Select Level
      </p>
      <label
        className="text"
      >
        Easy 
      </label>
      <input
        id="easy"
        name="difficulty"
        required={true}
        type="radio"
        value="easy"
      />
      <label
        className="text"
      >
        Medium 
      </label>
      <input
        id="medium"
        name="difficulty"
        type="radio"
        value="medium"
      />
      <label
        className="text"
      >
        Difficult 
      </label>
      <input
        id="difficult"
        name="difficulty"
        type="radio"
        value="hard"
      />
      <br />
      <select
        id="category"
        name="category"
        required={true}
        value={null}
      >
        <option
          name="Please Select Category"
          value=""
        >
           Please Select Category
        </option>
        <option
          name="Celebrity"
          value="26"
        >
          Celebrity
        </option>
        <option
          name="Mythology"
          value="20"
        >
          Mythology
        </option>
        <option
          name="History"
          value="23"
        >
          History
        </option>
        <option
          name="Animals"
          value="27"
        >
          Animals
        </option>
        <option
          name="Science & Computers"
          value="18"
        >
          Science & Computers
        </option>
      </select>
      <div
        className="buttonDiv"
      >
        <button
          className="playbutton"
          type="submit"
        >
          Play
        </button>
      </div>
    </form>
  </div>
</div>
`;
